src/components/VideoPreview.tsx

```diff
@@ -1,36 +1,78 @@
-} & VideoPreviewBaseMetaProps
+} & VideoPreviewBaseMetaProps &
+  Pick<VideoPreviewBaseProps, 'progress' | 'isLoading' | 'className'>
 
-const VideoPreview: React.FC<VideoPreviewProps> = ({ id, progress, className, ...metaProps }) => {
-  const { video, loading } = useVideo(id ?? '', { fetchPolicy: 'cache-first', skip: !id })
-  const isLoading = loading || id === undefined
+const VideoPreview: React.FC<VideoPreviewProps> = ({ id, isLoading = false, ...metaProps }) => {
+  const { video, internalIsLoadingState, videoHref } = useVideoSharedLogic(id, isLoading)
   return (
     <VideoPreviewBase
+      publisherMode={false}
       title={video?.title}
       channelHandle={video?.channel.handle}
       channelAvatarUrl={video?.channel.avatarPhotoUrl}
       createdAt={video?.createdAt}
       duration={video?.duration}
       views={video?.views}
       thumbnailUrl={video?.thumbnailUrl}
-      videoHref={id ? routes.video(id) : undefined}
+      videoHref={videoHref}
       channelHref={id ? routes.channel(video?.channel.id) : undefined}
-      progress={progress}
-      isLoading={isLoading}
-      className={className}
-      // TODO: handle menu contextMenuCallbacks
-      // contextMenuCallbacks={{}}
+      isLoading={internalIsLoadingState}
+      contentKey={id}
       {...metaProps}
     />
   )
 }
 
 export default VideoPreview
+
+export type VideoPreviewWPublisherProps = VideoPreviewProps &
+  Omit<VideoPreviewPublisherProps, 'publisherMode' | 'videoPublishState'>
+export const VideoPreviewPublisher: React.FC<VideoPreviewWPublisherProps> = ({
+  id,
+  isLoading = false,
+  isDraft,
+  ...metaProps
+}) => {
+  const { video, internalIsLoadingState, videoHref } = useVideoSharedLogic(id, isLoading)
+  const { drafts } = useDrafts('video')
+  const draft = id ? drafts.find((draft) => draft.id === id) : undefined
+  return (
+    <VideoPreviewBase
+      publisherMode
+      title={isDraft ? draft?.title : video?.title}
+      channelHandle={video?.channel.handle}
+      channelAvatarUrl={video?.channel.avatarPhotoUrl}
+      createdAt={isDraft ? new Date(draft?.updatedAt ?? '') : video?.createdAt}
+      duration={video?.duration}
+      views={video?.views}
+      thumbnailUrl={video?.thumbnailUrl}
+      videoHref={videoHref}
+      channelHref={id ? routes.channel(video?.channel.id) : undefined}
+      isLoading={internalIsLoadingState}
+      onCopyVideoURLClick={isDraft ? undefined : () => copyToClipboard(videoHref ? location.origin + videoHref : '')}
+      videoPublishState={video?.isPublic || video?.isPublic === undefined ? 'default' : 'unlisted'}
+      isDraft={isDraft}
+      contentKey={id}
+      {...metaProps}
+    />
+  )
+}
+
+const useVideoSharedLogic = (id?: string, isLoading?: boolean) => {
+  const { video, loading } = useVideo(id ?? '', { skip: !id })
+  const internalIsLoadingState = loading || !id || isLoading
+  const videoHref = id ? routes.video(id) : undefined
+  return { video, internalIsLoadingState, videoHref }
```

Comment:
Maybe let's just return `internalIsLoadingState` as loading from here? Name can be a bit misleading